tox.ini

```diff
@@ -9,7 +9,14 @@ allowlist_externals =
 deps =
     # -U keeps all requirements up-to-date
     -Urrequirements.txt
-    -Urrequirements_dev.txt
+    # All test cases require pytest
+    pytest
+    pytest-cov
+    pytest-sugar
+    hypothesis
+    coverage
+    # If we're in pypy don't install all of these dependencies, otherwise grab the full set.
+    !pypy37: -Urrequirements_dev.txt

```

Comment:
Do you know why it's necessary to exclude PyPy here?

Reply:
`mypy` doesn't support `pypy`. It's because the dependency `typed_ast` doesn't support `pypy`.